src/main/java/com/royalrangers/service/achievement/PlatoonAchievementService.java

```diff
@@ -234,17 +157,28 @@ public PlatoonRewardDto buildUserReward(UserReward item) {
 
         platoonRewardDto.setRewardId(item.getId());
         platoonRewardDto.setRewardType(item.getReward().getRewardType());
-        if (item.getReward().getRewardType().equals(RewardType.MEDAL)) {
-            platoonRewardDto.setCount(findAllMedalRewards().size());
-        } else if (item.getReward().getRewardType().equals(RewardType.LATH)) {
-            platoonRewardDto.setCount(findAllLathRewards().size());
-        } else if (item.getReward().getRewardType().equals(RewardType.STAR)) {
-            platoonRewardDto.setCount(findAllStarRewards().size());
-        } else if (item.getReward().getRewardType().equals(RewardType.CAMP)) {
-            platoonRewardDto.setCount(findAllCampRewards().size());
-        } else if (item.getReward().getRewardType().equals(RewardType.TRIP)) {
-            platoonRewardDto.setCount(findAllTripRewards().size());
+        if (item.getReward().getRewardType().equals(RewardType.MEDAL) && item.getAchievementState().equals(AchievementState.APPROVED)) {
```

Comment:
Please rewrite all enum comparisons to:
`<enum value>.equals(<value to compare>)`

Here you should write:
`RewardType.MEDAL.equals(item.getReward().getRewardType())`